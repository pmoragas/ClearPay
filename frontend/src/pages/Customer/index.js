import React, {useEffect} from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import { getCustomers } from 'store/customer/actions';
import BootstrapTable from 'react-bootstrap-table-next';
import ToolkitProvider, { Search } from 'react-bootstrap-table2-toolkit';

import styles from './styles.module.scss';
import { CUSTOMER_DETAIL_PATH } from 'router/paths';

const Character = () => {
    const { SearchBar } = Search;
    const dispatch = useDispatch();
    const { customers } = useSelector((state) => state.customer);

    const columns = [
        {
            dataField: 'id',
            text: 'Customer ID',
            headerStyle: {
				width: '15%',
				textAlign: 'left',
				textTransform: 'none',
			},
        },
        {
            dataField: 'first_name',
            text: 'First Name',
            headerStyle: {
				width: '20%',
				textAlign: 'left',
				textTransform: 'none',
			},
        },
        {
            dataField: 'last_name',
            text: 'Last Name',
            headerStyle: {
				width: '20%',
				textAlign: 'left',
				textTransform: 'none',
			},
        },
        {
            dataField: 'email',
            text: 'Email',
            headerStyle: {
				width: '30%',
				textAlign: 'left',
				textTransform: 'none',
			},
        },
        {
            dataField: 'link',
			text: 'Link',
			headerStyle: {
				width: '15%',
				textAlign: 'center',
				textTransform: 'none',
			},
			formatter: (rowContent, row) => {
				return (
					<Link
						className={styles.link}
						to={CUSTOMER_DETAIL_PATH.replace(':detailId', row.id)}
					>
						Open Customer
					</Link>
				);
			},
            align: 'center',
        }
    ];

    useEffect(() => {
        dispatch(getCustomers());
    }, [dispatch]);

    return (
        <div className={styles.container}>
            <ToolkitProvider
            keyField="id"
            data={ customers }
            columns={ columns }
            search
            >
            {
                props => (
                <div>
                    <SearchBar { ...props.searchProps } srText=''/>
                    <hr />
                    <BootstrapTable
                    { ...props.baseProps }

                    style={{widht:'100%'}}
                    rowStyle={ {height: '30px'} }
                    />
                </div>
                )
            }
            </ToolkitProvider>
        </div>
    );
}

export default Character;
