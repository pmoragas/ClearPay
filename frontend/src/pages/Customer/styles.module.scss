@import '~styles/colors';

.container {
    font-family: sans-serif;
    padding: 30px 30px;
    display: grid;
    grid-template-columns: repeat( auto-fit, minmax(500px, 1fr) );

    table {
      width: 100%;
    }

    .link {
      text-decoration: inherit;
      color: $tertiary;
      font-weight: 600;
      font-size: 14px;
      border-radius: .75rem;
      background-color: rgba($tertiary, 1);
      color: $white;
      padding: 5px;
      white-space: nowrap;

      &:visited {
        text-decoration: inherit;
      }

      &:hover {
        opacity: 50%;
      }

      &:active {
        text-decoration: inherit;
      }
    }


}