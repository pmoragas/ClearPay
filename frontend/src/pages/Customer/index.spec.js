import React from 'react';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import { render, screen } from '@testing-library/react';
import Customer from './index';
const { customers } = require('mock/customers');

let mockCustomerSelector;

jest.mock('react-redux', () => ({
	useSelector: () => mockCustomerSelector,
    useDispatch: () => {
		return jest.fn();
	},
}));

describe('Customer', () => {
	test('Render Customer Page', () => {
        const history = createMemoryHistory();
        mockCustomerSelector = { customers };
		render(
            <Router history={ history }>
                <Customer/>
            </Router>);

		expect(screen.getAllByText('Open Customer').length).toEqual(20);
		expect(screen.getByText('Adam')).toBeInTheDocument();
		expect(screen.getByText('Bank')).toBeInTheDocument();
		expect(screen.getByText('ibrockiee@mediafire.com')).toBeInTheDocument();
		expect(screen.getByText('20')).toBeInTheDocument();
	});

	test('snapshot is ok', () => {
        const history = createMemoryHistory();
        mockCustomerSelector = {customers};

        const { container } = render(
            <Router history={ history }>
                <Customer/>
            </Router>);

		expect(container).toMatchSnapshot();
	});
});
