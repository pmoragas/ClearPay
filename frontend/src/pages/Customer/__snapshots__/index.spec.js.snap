// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Customer snapshot is ok 1`] = `
<div>
  <div
    class="container"
  >
    <div>
      <label
        class="search-label"
        for="search-bar-0"
      >
        <span
          class="sr-only"
          id="search-bar-0-label"
        />
        <input
          aria-labelledby="search-bar-0-label"
          class="form-control "
          id="search-bar-0"
          placeholder="Search"
          type="text"
          value=""
        />
      </label>
      <hr />
      <div
        class="react-bootstrap-table"
      >
        <table
          class="table table-bordered"
        >
          <thead>
            <tr>
              <th
                style="width: 15%; text-align: left; text-transform: none;"
                tabindex="0"
              >
                Customer ID
              </th>
              <th
                style="width: 20%; text-align: left; text-transform: none;"
                tabindex="0"
              >
                First Name
              </th>
              <th
                style="width: 20%; text-align: left; text-transform: none;"
                tabindex="0"
              >
                Last Name
              </th>
              <th
                style="width: 30%; text-align: left; text-transform: none;"
                tabindex="0"
              >
                Email
              </th>
              <th
                style="width: 15%; text-align: center; text-transform: none;"
                tabindex="0"
              >
                Link
              </th>
            </tr>
          </thead>
          <tbody>
            <tr
              style="height: 30px;"
            >
              <td>
                1
              </td>
              <td>
                Jan
              </td>
              <td>
                Stannard
              </td>
              <td>
                jstannard0@g.co
              </td>
              <td
                style="text-align: center;"
              >
                <a
                  class="link"
                  href="/customer/1"
                >
                  Open Customer
                </a>
              </td>
            </tr>
            <tr
              style="height: 30px;"
            >
              <td>
                2
              </td>
              <td>
                Nichole
              </td>
              <td>
                Bultitude
              </td>
              <td>
                nbultitude1@fastcompany.com
              </td>
              <td
                style="text-align: center;"
              >
                <a
                  class="link"
                  href="/customer/2"
                >
                  Open Customer
                </a>
              </td>
            </tr>
            <tr
              style="height: 30px;"
            >
              <td>
                3
              </td>
              <td>
                Cornela
              </td>
              <td>
                Litel
              </td>
              <td>
                clitel2@t-online.de
              </td>
              <td
                style="text-align: center;"
              >
                <a
                  class="link"
                  href="/customer/3"
                >
                  Open Customer
                </a>
              </td>
            </tr>
            <tr
              style="height: 30px;"
            >
              <td>
                4
              </td>
              <td>
                Ulises
              </td>
              <td>
                Moan
              </td>
              <td>
                umoan3@yahoo.co.jp
              </td>
              <td
                style="text-align: center;"
              >
                <a
                  class="link"
                  href="/customer/4"
                >
                  Open Customer
                </a>
              </td>
            </tr>
            <tr
              style="height: 30px;"
            >
              <td>
                5
              </td>
              <td>
                Broderick
              </td>
              <td>
                Wraith
              </td>
              <td>
                bwraith4@google.nl
              </td>
              <td
                style="text-align: center;"
              >
                <a
                  class="link"
                  href="/customer/5"
                >
                  Open Customer
                </a>
              </td>
            </tr>
            <tr
              style="height: 30px;"
            >
              <td>
                6
              </td>
              <td>
                Andrew
              </td>
              <td>
                Newlan
              </td>
              <td>
                anewlan5@chron.com
              </td>
              <td
                style="text-align: center;"
              >
                <a
                  class="link"
                  href="/customer/6"
                >
                  Open Customer
                </a>
              </td>
            </tr>
            <tr
              style="height: 30px;"
            >
              <td>
                7
              </td>
              <td>
                Izak
              </td>
              <td>
                Bowsher
              </td>
              <td>
                ibowsher6@cloudflare.com
              </td>
              <td
                style="text-align: center;"
              >
                <a
                  class="link"
                  href="/customer/7"
                >
                  Open Customer
                </a>
              </td>
            </tr>
            <tr
              style="height: 30px;"
            >
              <td>
                8
              </td>
              <td>
                Adam
              </td>
              <td>
                Giannazzo
              </td>
              <td>
                agiannazzo7@yahoo.co.jp
              </td>
              <td
                style="text-align: center;"
              >
                <a
                  class="link"
                  href="/customer/8"
                >
                  Open Customer
                </a>
              </td>
            </tr>
            <tr
              style="height: 30px;"
            >
              <td>
                9
              </td>
              <td>
                Darrick
              </td>
              <td>
                Romer
              </td>
              <td>
                dromer8@ihg.com
              </td>
              <td
                style="text-align: center;"
              >
                <a
                  class="link"
                  href="/customer/9"
                >
                  Open Customer
                </a>
              </td>
            </tr>
            <tr
              style="height: 30px;"
            >
              <td>
                10
              </td>
              <td>
                Illa
              </td>
              <td>
                Bank
              </td>
              <td>
                ibank9@surveymonkey.com
              </td>
              <td
                style="text-align: center;"
              >
                <a
                  class="link"
                  href="/customer/10"
                >
                  Open Customer
                </a>
              </td>
            </tr>
            <tr
              style="height: 30px;"
            >
              <td>
                11
              </td>
              <td>
                Joanie
              </td>
              <td>
                McClifferty
              </td>
              <td>
                jmccliffertya@example.com
              </td>
              <td
                style="text-align: center;"
              >
                <a
                  class="link"
                  href="/customer/11"
                >
                  Open Customer
                </a>
              </td>
            </tr>
            <tr
              style="height: 30px;"
            >
              <td>
                12
              </td>
              <td>
                Skippy
              </td>
              <td>
                Ber
              </td>
              <td>
                sberb@simplemachines.org
              </td>
              <td
                style="text-align: center;"
              >
                <a
                  class="link"
                  href="/customer/12"
                >
                  Open Customer
                </a>
              </td>
            </tr>
            <tr
              style="height: 30px;"
            >
              <td>
                13
              </td>
              <td>
                Ram
              </td>
              <td>
                Lewsie
              </td>
              <td>
                rlewsiec@theatlantic.com
              </td>
              <td
                style="text-align: center;"
              >
                <a
                  class="link"
                  href="/customer/13"
                >
                  Open Customer
                </a>
              </td>
            </tr>
            <tr
              style="height: 30px;"
            >
              <td>
                14
              </td>
              <td>
                Ronalda
              </td>
              <td>
                Foukx
              </td>
              <td>
                rfoukxd@ftc.gov
              </td>
              <td
                style="text-align: center;"
              >
                <a
                  class="link"
                  href="/customer/14"
                >
                  Open Customer
                </a>
              </td>
            </tr>
            <tr
              style="height: 30px;"
            >
              <td>
                15
              </td>
              <td>
                Immanuel
              </td>
              <td>
                Brockie
              </td>
              <td>
                ibrockiee@mediafire.com
              </td>
              <td
                style="text-align: center;"
              >
                <a
                  class="link"
                  href="/customer/15"
                >
                  Open Customer
                </a>
              </td>
            </tr>
            <tr
              style="height: 30px;"
            >
              <td>
                16
              </td>
              <td>
                Fair
              </td>
              <td>
                Godsal
              </td>
              <td>
                fgodsalf@gov.uk
              </td>
              <td
                style="text-align: center;"
              >
                <a
                  class="link"
                  href="/customer/16"
                >
                  Open Customer
                </a>
              </td>
            </tr>
            <tr
              style="height: 30px;"
            >
              <td>
                17
              </td>
              <td>
                Rozanna
              </td>
              <td>
                Wafer
              </td>
              <td>
                rwaferg@vkontakte.ru
              </td>
              <td
                style="text-align: center;"
              >
                <a
                  class="link"
                  href="/customer/17"
                >
                  Open Customer
                </a>
              </td>
            </tr>
            <tr
              style="height: 30px;"
            >
              <td>
                18
              </td>
              <td>
                Deck
              </td>
              <td>
                Goodlad
              </td>
              <td>
                dgoodladh@examiner.com
              </td>
              <td
                style="text-align: center;"
              >
                <a
                  class="link"
                  href="/customer/18"
                >
                  Open Customer
                </a>
              </td>
            </tr>
            <tr
              style="height: 30px;"
            >
              <td>
                19
              </td>
              <td>
                Weber
              </td>
              <td>
                Ladewig
              </td>
              <td>
                wladewigi@i2i.jp
              </td>
              <td
                style="text-align: center;"
              >
                <a
                  class="link"
                  href="/customer/19"
                >
                  Open Customer
                </a>
              </td>
            </tr>
            <tr
              style="height: 30px;"
            >
              <td>
                20
              </td>
              <td>
                Stanislaus
              </td>
              <td>
                Gudgen
              </td>
              <td>
                sgudgenj@nhs.uk
              </td>
              <td
                style="text-align: center;"
              >
                <a
                  class="link"
                  href="/customer/20"
                >
                  Open Customer
                </a>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;
